clang/include/clang/AST/DeclTemplate.h

```diff
@@ -2596,27 +2588,14 @@ class VarTemplateSpecializationDecl : public VarDecl,
   llvm::PointerUnion<VarTemplateDecl *, SpecializedPartialSpecialization *>
   SpecializedTemplate;
 
-  /// Further info for explicit template specialization/instantiation.
-  struct ExplicitSpecializationInfo {
-    /// The type-as-written.
-    TypeSourceInfo *TypeAsWritten = nullptr;
-
-    /// The location of the extern keyword.
-    SourceLocation ExternLoc;
-
-    /// The location of the template keyword.
-    SourceLocation TemplateKeywordLoc;
-
-    ExplicitSpecializationInfo() = default;
-  };
-
   /// Further info for explicit template specialization/instantiation.
   /// Does not apply to implicit specializations.
-  ExplicitSpecializationInfo *ExplicitInfo = nullptr;
+  llvm::PointerUnion<const ASTTemplateArgumentListInfo *,
```

Comment:
Can we extract this type somewhere universal via a `using` declaration and use it via a useful name?

Reply:
Its to give it a 'useful' descriptive name.  The pointer-union is necessarily opaque as to its meaning, so a name that reflects the name is really helpful when reading the code.

Reply:
Alright. Do you have any suggestions for the name?

Reply:
Not a good one :)  Perhpas: `TemplateArgListInfoOrExplicitSpecInfo` ?